### DISCUSSION TITLE
How to access item properties in the PDF emitter?

### DISCUSSION THREAD

**Discussion by hvbargen** [eclipse-birt/birt#1891] [Ideas]
In my vacation, I'm experimenting with creating tagged PDFs.
There are a lot of questions, but I'd like to start with a simple prototype.

I know the report items have a "tagType" property and this can be queried by calling the getTagType method.

But I don't have a clue how to access the items from inside the emitter to read the property!

For example, there's a ReportItemHandleImpl class which supports the method, but if I look into the PDFPageDevice and PDFRender classes etc with the debugger, it seems I cannot access them.

Actually, the algorithm that BIRT uses to create a paginated layout from the design is a mystery to me.

Is it documentend anywhere or can anyone give me some clues?

**Comment by hvbtup** [comment]
I finally found a way myself.
It's complicated.
For those interested, look at my (hvbargen) first few changes in the tagged-pdf branch at hvbargen/birt or triestram-partner.birt.
There I solved getting the PDF tag type.

**Reply by merks** [reply]
Sorry for not helping.  It's been a bit crazy busy with travel and with the SimRel release.
